Implemented the MyClubs Screen (UI + ViewModel)

nirvan73 opened the pull request:
## 📝Overview
- This PR fixes or fixes part of https://github.com/bsoc-bitbyte/Nudj/issues/38
- This PR does the following: Implements the my clubs screen with its (UI + ViewModel) based on the provided Figma design , while keeping the colors , theme , fonts and shapes in mind using the follow Repository and User Repository.

---

## ✅ Checklist
✅ My code follows the project's style and structure
✅ Matches Figma design specifications
✅ Tested locally with no errors

---

## 📸 Proof that changes are correct
- Added screen recording and the screenshots of the my clubs page layout for any UI changes/feedback.
- Also the previews for the composable functions implemented inside the MyClubsScreen.kt files are made inside the file itself.

---

## 🗒️ Additional Notes
- Added the required strings in string.xml files.
- Registered the MyClubs screen in the MainActivity.kt and the Screens.kt both.
- Added the circular progress indicator between each process to ensure a smooth and seamless user experience.


https://github.com/user-attachments/assets/71733116-06e5-43f6-b036-e96719164d15

<img width="736" height="837" alt="Screenshot 2025-07-14 114511" src="https://github.com/user-attachments/assets/f594b88c-f115-42f1-9214-bef03b127839" />

hemang-mishra commented on app/src/main/java/com/tpc/nudj/ui/screens/myClubs/MyClubsViewModel.kt:
```diff
@@ -0,0 +1,141 @@
+class MyClubsViewModel @Inject constructor(
+    private val userRepository: UserRepository,
+    private val followRepository: FollowRepository
+) : ViewModel() {
```
We want some error handling as well. So we need to propagate the errors the screens and not just log them.
So either extend the viewmodel from BaseViewModel which already have error handling setup. But you will have to create ui state for that, which would be a lot of change to the code.
So for this screen, you have add a separate flow for propagating errors to ui. Collect them in screen and use snackbar do display the error.
But in future try to use BaseViewModel as it reduces some boilerplate code for complex screens.

hemang-mishra commented on app/src/main/java/com/tpc/nudj/ui/screens/myClubs/MyClubsScreen.kt:
```diff
@@ -0,0 +1,511 @@
+        onBack = { onBackClicked() },
+        isSaved = isSaved,
+        onCancelClub = { viewModel.removeSelectedClub(it) },
+        onSaveClicked = { viewModel.onFollowClubs() }
```
When the saving is successful, better navigate back to previous screen. Because they would click save only when they have finished their work.

nirvan73 commented:
Updated the files as per asked changes , added backNavigation on onSaved so that the user gets back to the previous screen after saving the followed clubs list and also created a MyClubsUiState to manage the loading , toastMessage and similar UI feedback. Used snackbar to show the toast messages. Here is the video preview:- 


https://github.com/user-attachments/assets/3e8b700b-b2ed-40a7-ad61-4899f87327d6
